### ISSUE TITLE
Plugin System breaks for certain command names

### ISSUE THREAD

**Issue by Bowserinator** [BWBellairs/Athena#29]
The current add command method looks like this:
```javascript
add_cmd(name, func) {
        this[name] = func;
}
```

However, one could easily break the plugin class by naming the command something that already exists as a property in the Plugin class, such as "bot", "add_cmd", etc... as it will override the existing property/method with an IRC command.

I haven't gotten around to looking at how Plugins are used in other files, but this could be an easy fix by doing 

```javascript
this.commands[name] = func;
```

or something like that.

**Comment by wolfy1339** [comment]
I actually thought of doing that when we were implementing hooks

**Comment by wolfy1339** [comment]
Closed by fb63e62fff7d2ce3c6ea4ace4a458b106562426c
